tests/acceptance/AdminDatasetForm.feature

```diff
@@ -291,6 +291,18 @@ Feature: form to update dataset details
     Then I am on "dataset/100006"
     And I should see "bam"
 
+  @ok @issue-2061
+  Scenario: Can delete all keywords on update
+    Given I am on "/adminDataset/update/id/8"
+    And I click on keywords field
+    And I fill in keywords fields with "bam"
+    And I press the button "Save"
+    When I am on "/adminDataset/update/id/8"
+    And I click on delete keyword button
+    And I press the button "Save"
+    Then I am on "dataset/100006"
+    And I should not see "bam"
+
```

Comment:
Hi @luistoptal,

This test is failing on my side:
```
 % docker-compose run --rm codecept run --no-redirect -g issue-2061 acceptance tests/acceptance/AdminDatasetForm.feature -vv

Codeception PHP Testing Framework v5.0.4 https://helpukrainewin.org
[Groups] issue-2061 

  [Connecting To Db] {"config":{"populate":true,"cleanup":true,"reconnect":false,"waitlock":0,"dump":"sql/gigadb.pgdmp","populator":"PGPASSWORD=$password pg_restore -h database -p $port -U gigadb -d gigadb --no-owner /var/www/sql/gigadb.pgdmp","skip_cleanup_if_failed":false,"dsn":"pgsql:host=database;dbname=gigadb;port=5432","user":"gigadb","password":""},"options":[]}
  [Db] Connected to default gigadb
  [Db] Executing Populator: `PGPASSWORD= pg_restore -h database -p 5432 -U gigadb -d gigadb --no-owner /var/www/sql/gigadb.pgdmp`
  [Db] Populator Finished.

Acceptance Tests (1) --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Modules: \Helper\Acceptance, Asserts, Db, WebDriver
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
form to update dataset details: Can delete all keywords on update
Signature: AdminDatasetForm:Can delete all keywords on update
Test: tests/acceptance/AdminDatasetForm.feature:Can delete all keywords on update
Scenario --
  [Db] Executing Populator: `PGPASSWORD= pg_restore -h database -p 5432 -U gigadb -d gigadb --no-owner /var/www/sql/gigadb.pgdmp`
  [Db] Populator Finished.
 As a curator
 I want a form to update dataset details
 So that the dataset information is up-to-date
 Given i have signed in as admin 
   I am on url "http://gigadb.test"
  [Host] http://gigadb.test
   I am on page "/site/login"
  [GET] http://gigadb.test/site/login
   I fill field {"name":"LoginForm[username]"},"admin@gigadb.org"
   I fill field {"name":"LoginForm[password]"},"gigadb"
   I click "Login"
   I wait for text "Home",10
 Given i am on "/admin dataset/update/id/8" 
   I am on page "/adminDataset/update/id/8"
  [GET] http://gigadb.test/adminDataset/update/id/8
 And i click on keywords field 
   I click {"css":".placeholder"}
 And i fill in keywords fields with "bam" 
   I type "bam"
   I click with left button {"css":"#urltoredirect"}
   I wait for text "bam",5,".tag-editor-tag"
 And i press the button "save" 
   I click "Save"
 When i am on "/admin dataset/update/id/8" 
   I am on page "/adminDataset/update/id/8"
  [GET] http://gigadb.test/adminDataset/update/id/8
 And i click on delete keyword button 
   I click ".tag-editor-delete"
 And i press the button "save" 
   I click "Save"
 Then i am on "dataset/100006" 
   I am on page "dataset/100006"
  [GET] http://gigadb.test/dataset/100006
 And i should not see "bam" 
   I don't see "bam"
  Screenshot and page source were saved into '/project/tests/_output/' dir
 FAIL 
```

And you can see the below screenshot that `bam` is still on the dataset page:

![AdminDatasetForm Can delete all keywords on update fail](https://github.com/user-attachments/assets/5dc29d3c-b058-4c25-9ef3-04bc9b85fca6)

Reply:
Hi @luistoptal,

Regarding to this issue, here is my findings:
1. The dataset view page is cached, and the cache will be updated based on the `creation_date` column in the curation_log table or the `created_at` column in the dataset_log table
2. The update of the keywords will be saved into the curation_log table with date format `Y-m-d`, which obviously not precise enough to distinguish whether the cache should be updated or not if the changes were made within the same day

**issue:** The date format of the`creation_date` column in the curation_log table is not precise enough
**suggestion:** Update the date format to `"Y-m-d H:i:s"` in protected/models/CurationLog.php:95` as below:
```
$curationlog->creation_date = date("Y-m-d H:i:s");
```

 After the above refactoring, the acceptance should be passed and the keywords in the view page should also be consistent with the admin page.

```
% docker-compose run --rm codecept run --no-redirect -g issue-2061 acceptance tests/acceptance/AdminDatasetForm.feature -vv

Codeception PHP Testing Framework v5.0.4 https://helpukrainewin.org
[Groups] issue-2061 

  [Connecting To Db] {"config":{"populate":true,"cleanup":true,"reconnect":false,"waitlock":0,"dump":"sql/gigadb.pgdmp","populator":"PGPASSWORD=$password pg_restore -h database -p $port -U gigadb -d gigadb --no-owner /var/www/sql/gigadb.pgdmp","skip_cleanup_if_failed":false,"dsn":"pgsql:host=database;dbname=gigadb;port=5432","user":"gigadb","password":""},"options":[]}
  [Db] Connected to default gigadb
  [Db] Executing Populator: `PGPASSWORD= pg_restore -h database -p 5432 -U gigadb -d gigadb --no-owner /var/www/sql/gigadb.pgdmp`
  [Db] Populator Finished.

Acceptance Tests (1) --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Modules: \Helper\Acceptance, Asserts, Db, WebDriver
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
form to update dataset details: Can delete all keywords on update
Signature: AdminDatasetForm:Can delete all keywords on update
Test: tests/acceptance/AdminDatasetForm.feature:Can delete all keywords on update
Scenario --
  [Db] Executing Populator: `PGPASSWORD= pg_restore -h database -p 5432 -U gigadb -d gigadb --no-owner /var/www/sql/gigadb.pgdmp`
  [Db] Populator Finished.
 As a curator
 I want a form to update dataset details
 So that the dataset information is up-to-date
 Given i have signed in as admin 
   I am on url "http://gigadb.test"
  [Host] http://gigadb.test
   I am on page "/site/login"
  [GET] http://gigadb.test/site/login
   I fill field {"name":"LoginForm[username]"},"admin@gigadb.org"
   I fill field {"name":"LoginForm[password]"},"gigadb"
   I click "Login"
   I wait for text "Home",10
 Given i am on "/admin dataset/update/id/8" 
   I am on page "/adminDataset/update/id/8"
  [GET] http://gigadb.test/adminDataset/update/id/8
 And i click on keywords field 
   I click {"css":".placeholder"}
 And i fill in keywords fields with "bam" 
   I type "bam"
   I click with left button {"css":"#urltoredirect"}
   I wait for text "bam",5,".tag-editor-tag"
 And i press the button "save" 
   I click "Save"
 When i am on "/admin dataset/update/id/8" 
   I am on page "/adminDataset/update/id/8"
  [GET] http://gigadb.test/adminDataset/update/id/8
 And i click on delete keyword button 
   I click ".tag-editor-delete"
 And i press the button "save" 
   I click "Save"
 Then i am on "dataset/100006" 
   I am on page "dataset/100006"
  [GET] http://gigadb.test/dataset/100006
 And i should not see "bam" 
   I don't see "bam"
 PASSED 

-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Time: 00:57.195, Memory: 12.00 MB

OK (1 test, 1 assertion)
  [Db] Disconnected from default
```